### ISSUE TITLE
Hello, how do I use three pseudo labels to train?

### ISSUE THREAD

**Issue by ma3252788** [DUTyimmy/MFNet#1]
I want to train simultaneously with three pseudo-labels or more fake labels. How should I modify the code?

Thank you！

**Comment by DUTyimmy** [comment]
Thanks for your attention! You can copy and add the code of **directive filter** in model_densenet.py for corresponding your specified number of pseudo labels.  And don't forget to make the appropriate changes in the loss term in trainsal.py.

**Comment by ma3252788** [comment]
Thank you, may I only need to modify these two files?

> Thanks for your attention! You can copy and add the code of **directive filter** in model_densenet.py for corresponding your specified number of pseudo labels. And don't forget to make the appropriate changes in the loss term in trainsal.py.

Thank you, may I only need to modify these two files?

In addition, if there are more than one, how to calculate the loss is better? For example, there are three data sets, which are A, B, and C. and loss= A-B-C ?

**Comment by DUTyimmy** [comment]
1. You need to modify dataloader.py for loading other pseudo labels, and there also may be some minor problems with other files, you can make changes accordingly according to the debugging situation.
2. If you need three pseudo labels, just take those labels **equally**. Such as the the final multi-guidance loss term: BCEloss(saliency prediction, (the average outputs of three DFs after refinement)). You could also try to find out the effect of different weights of DFs to the final performance, i think it will be a interesting issue.
